core/management/commands/import_data.py

```diff
@@ -55,96 +47,14 @@ def handle(self, *args, **kwargs):
-                    encoding=encoding,
-                    dialect="excel",
-                    database_uri=database_uri,
-                    table_name=table_name,
-                    create_table=False,
-                    timeout=timeout,
-                    callback=progress.update,
-                    schema=schema,
-                )
-            except RuntimeError as exception:
-                progress.close()
-                print("ERROR: {}".format(exception.args[0]))
-                exit(1)
-            else:
-                progress.close()
-                table.import_date = timezone.now()
-                table.save()
-                if collect_date:
-                    table.version.collected_at = collect_date
-                    table.version.save()
-                end_time = time.time()
-                duration = end_time - start_time
-                rows_imported = import_meta["rows_imported"]
-                print(
-                    "  done in {:7.3f}s ({} rows imported, {:.3f} rows/s).".format(
-                        duration, rows_imported, rows_imported / duration
-                    )
-                )
-            Model = table.get_model(cache=False)
-            table.invalidate_cache()
-
-        if vacuum:
-            print("Running VACUUM ANALYSE...", end="", flush=True)
-            start = time.time()
-            Model.analyse_table()
-            end = time.time()
-            print("  done in {:.3f}s.".format(end - start))
-
-        if create_filter_indexes:
-            # TODO: warn if field has_choices but not in Table.filtering
-            print("Creating filter indexes...", end="", flush=True)
-            start = time.time()
-            Model.create_indexes()  # TODO: add "IF NOT EXISTS"
-            end = time.time()
-            print("  done in {:.3f}s.".format(end - start))
-
-        if fill_choices:
-            print("Filling choices...")
-            start = time.time()
-            choiceables = Field.objects.for_table(table).choiceables()
-            for field in choiceables:
-                print("  {}".format(field.name), end="", flush=True)
-                start_field = time.time()
-                field.update_choices()
-                field.save()
-                end_field = time.time()
-                print(" - done in {:.3f}s.".format(end_field - start_field))
-            end = time.time()
-            print("  done in {:.3f}s.".format(end - start))
-
-        if clear_view_cache:
-            print("Clearing view cache...")
-            cache.clear()
+        ImportDataCommand.execute(
+            dataset_slug,
+            tablename,
+            filename,
+            import_data=import_data,
+            vacuum=vacuum,
+            clear_view_cache=clear_view_cache,
+            create_filter_indexes=create_filter_indexes,
+            fill_choices=fill_choices,
+            collect_date=collect_date,
+        )
```

Comment:
Seguindo a sugestão de refatoração do método:

```suggestion
        import_table = ImportDataCommand.from_tablename(dataset_slug, tablename)
        import_table.execute(
            filename,
            import_data=import_data,
            vacuum=vacuum,
            clear_view_cache=clear_view_cache,
            create_filter_indexes=create_filter_indexes,
            fill_choices=fill_choices,
            collect_date=collect_date,
        )
```